pkg/volume/csi/csi_plugin.go

```diff
@@ -51,28 +51,33 @@ const (
 	// the unix domain socket path for each installed csi driver.
 	// TODO (vladimirvivien) would be nice to name socket with a .sock extension
 	// for consistency.
-	csiAddrTemplate = "/var/lib/kubelet/plugins/%v/csi.sock"
-	csiTimeout      = 15 * time.Second
-	volNameSep      = "^"
-	volDataFileName = "vol_data.json"
-	fsTypeBlockName = "block"
+	csiAddrTemplate   = "/var/lib/kubelet/plugins/%v/csi.sock"
+	csiDefaultTimeout = 15 * time.Second
+	volNameSep        = "^"
+	volDataFileName   = "vol_data.json"
+	fsTypeBlockName   = "block"
 
 	// TODO: increase to something useful
 	csiResyncPeriod = time.Minute
+
+	volumeKind           = "Volume"
+	persistentVolumeKind = "PersistentVolume"
 )
 
 type csiPlugin struct {
-	host              volume.VolumeHost
-	blockEnabled      bool
-	csiDriverLister   csilister.CSIDriverLister
-	csiDriverInformer csiinformer.CSIDriverInformer
+	host                 volume.VolumeHost
+	blockEnabled         bool
+	inlineFeatureEnabled bool
+	csiDriverLister      csilister.CSIDriverLister
+	csiDriverInformer    csiinformer.CSIDriverInformer
 }
 
 // ProbeVolumePlugins returns implemented plugins
 func ProbeVolumePlugins() []volume.VolumePlugin {
 	p := &csiPlugin{
-		host:         nil,
-		blockEnabled: utilfeature.DefaultFeatureGate.Enabled(features.CSIBlockVolume),
+		host:                 nil,
+		blockEnabled:         utilfeature.DefaultFeatureGate.Enabled(features.CSIBlockVolume),
+		inlineFeatureEnabled: utilfeature.DefaultFeatureGate.Enabled(features.CSIInlineVolume),
```

Comment:
We shouldn't be adding these feature gates as fields inside the plugin object. Wherever you need to check whether inline feature is enabled I would put `utilfeature.DefaultFeatureGate.Enabled(features.CSIInlineVolume)` directly. This makes it very clear the directive is coming from a feature gate which is just a temporary gate for functionality so that people dont "accidentally" use the feature. It's not a first class property of CSI